BedrockServer.cpp

```diff
@@ -788,6 +788,22 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
         }
     }
 
+    // If we're on a different version from leader AND this is not a command that was escalatedImmediatly, 
+    // we'll  return a 542 error so that the client knows it needs to call another server from the cluster. 
+    // We're doing this to prevent overloading leader with requests from all servers instead of spreading 
+    // them equaly between available servers.
+    if (state == SQLiteNodeState::FOLLOWING && _version != _leaderVersion.load() && !command->escalateImmediately && !command->complete) {
```

Comment:
Could we move `if (state == SQLiteNodeState::FOLLOWING &&  && !command->complete)` into a separate `if` block containing this and the above block?

Then we could do:
```
if (state == SQLiteNodeState::FOLLOWING &&  && !command->complete) {
    if (command->escalateImmediately) {
        runOnLeader();
    } else if (_version != _leaderVersion.load()) {
        runOnValidFollowerPeer();
    }
}
```

I think that would be a little bit easier to follow.

Reply:
Good suggestion, applied it